src/main/java/seedu/address/logic/parser/quiz/QuizAnswerParser.java

```diff
@@ -14,14 +18,17 @@
     /**
      * Used for initial separation of question index and answer.
      */
-    private static final Pattern BASIC_INPUT_FORMAT = Pattern.compile("(?<index>\\s+)(?<answer>)");
+    private static final Pattern BASIC_INPUT_FORMAT = Pattern.compile("(?<index>\\S+)(?<answer>.*)");
```

Comment:
The regex should be `(?<index>\\d+):(?<answer>.+)`. This enforces that the index is numeric and that the answer is non-empty.